lib/generic/optionally_set.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generic
+
+import "encoding/json"
+
+// OptionallySet allows distinguishing between "Missing Field" (Schema Cold Start)
+// and "Zero Value" (Valid Data).
+type OptionallySet[T any] struct {
+	Value T
+	IsSet bool
+}
+
+func (o OptionallySet[T]) IsZero() bool {
```

Comment:
What if the value is set to zero?

Reply:
If the caller sets IsTrue to true, we still print it out.

Background: We leverage the new omitzero [feature](https://go.dev/doc/go1.24#encodingjsonpkgencodingjson) which allows us to control the zero state of a field. If we declare it as "Zero" by setting IsSet to true, we assume that the field didn't exist beforehand.